bitswap-monitoring-client/src/prom.rs

```diff
@@ -89,32 +103,50 @@ pub(crate) struct MetricsByMonitorAndCountry {
     pub(crate) num_disconnected: GenericCounter<AtomicU64>,
 }
 
-impl MetricsByMonitorAndCountry {
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy)]
+pub enum OriginType {
+    Peer,
+    Gateway,
+}
+
+impl Metrics {
     /// Creates a set of metrics consisting of a few popular countries and the special
     /// error-condition countries, for the given monitor.
     pub(crate) fn create_basic_set(
         monitor_name: &str,
-    ) -> HashMap<String, MetricsByMonitorAndCountry> {
+    ) -> HashMap<(celes::Country, OriginType), Metrics> {
```

Comment:
I'd create a struct for the key type, that helps readability.

Also: Because of the performance considerations above, I'd leave this to use the alpha2 code of a country as key instead of the actual country. I know `Unknown` and `Error` are not alpha2 codes, but that's perfect, since we can never confuse them :) If you really want to separate them, we could also add an enum like so:
```rust
#[derive(all the things)]
pub enum MetricsCountry{
    Unknown,
    Error,
    Alpha2(String)
}
```